ChildProject/pipelines/metrics.py

```diff
@@ -20,26 +20,65 @@ def __init__(
         recordings: Union[str, List[str], pd.DataFrame] = None,
         from_time: str = None,
         to_time: str = None,
+        rec_cols: str = None,
+        child_cols: str = None,
     ):
 
         self.project = project
         self.am = ChildProject.annotations.AnnotationManager(self.project)
 
-        recording_columns = {
+        #necessary columns to construct the metrics
+        join_columns = {
             "recording_filename",
             "child_id",
             "duration",
             "session_id",
             "session_offset",
         }
-        recording_columns &= set(self.project.recordings.columns)
-
+        #get existing columns of the dataset for recordings
+        correct_cols = set(self.project.recordings.columns)
+        if by not in correct_cols: exit("<{}> is not specified in this dataset, cannot extract from it".format(by))
+        if rec_cols:
+            #when wanted columns specified, build the list and verify they exist (warn otherwise)
+            rec_cols=set(rec_cols.split(","))
+            for i in rec_cols:
+                if i not in correct_cols:
+                    print("Warning, specified column <{}> does not exist in recordings.csv, ignoring this column".format(i))
+            rec_cols &= correct_cols
+            #add wanted columns to the one we already get
+            join_columns.update(rec_cols)
+        self.rec_cols = rec_cols
+        
+        join_columns &= correct_cols
+        
+        #join dataset annotation with their info in recordings.csv
         self.am.annotations = self.am.annotations.merge(
-            self.project.recordings[recording_columns],
+            self.project.recordings[list(join_columns)],
             left_on="recording_filename",
             right_on="recording_filename",
         )
-
+        
+        #get existing columns of the dataset for children
+        correct_cols = set(self.project.children.columns)
+        if child_cols:
+            #when wanted columns specified, build the list and verify they exist (warn otherwise)
+            child_cols = set(child_cols.split(","))
+            child_cols.add("child_id")
+            for i in child_cols:
+                if i not in correct_cols:
+                    print("Warning, specified column <{}> does not exist in children.csv, ignoring this column".format(i))
```

Comment:
same comment as above RE warning

Reply:
`print("Warning, requested column <{}> does not exist in children.csv, ignoring this column. existing columns are : {}".format(i,correct_cols))`